### PR TITLE
Prac 06 feedback

### PR DESCRIPTION
**Pull request by bipinsubedi4**
@BenChisnell

### CODE REVIEW

File: prac_06/guitar.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        self.name = name
+        self.year = year
+        self.cost = cost
+
+    def get_age(self):
+        current_year = 2024
+        return current_year - self.year
+
+    def is_vintage(self):
+        return self.get_age() >=50
```
Comment (BenChisnell) [inline prac_06/guitar.py]: Vintage age is correct (50), but it would be better if it was a constant

File: prac_06/guitars.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+def main():
+    """Prompt user to input guitar details and display all entered guitars."""
+    guitars = []
+    print("My guitars!")
+
+    # Get the first input outside the loop to start the condition
+    name = input("Name: ")
+
+    while name:  # Continue as long as name is not blank
```
Comment (BenChisnell) [inline prac_06/guitars.py]: This while loop won't work as it just says "while name" and doesn't have a condition.
For it to works, it needs to be:
while name != ""

File: prac_06/guitars.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+        guitar = Guitar(name, year, cost)
+        guitars.append(guitar)
+        print(f"{guitar} added.\n")
+
+        # Prompt for the next guitar name
+        name = input("Name: ")
+
+    # Display all guitars
+    print("\nThese are my guitars:")
```
Comment (BenChisnell) [inline prac_06/guitars.py:33]: The "\n" doesn't need to be in there, as it's going to be printed on a new line from the print statement.

File: prac_06/programming_language.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+Start Time: 6:50 AM
+End Time: 7:07 AM, 17 minutes
+
+This module contains the ProgrammingLanguage class to store information
+about various programming languages, including their typing, reflection capability,
+and year of creation.
+"""
+class ProgrammingLanguage:
+    def __init__(self, language, typing, reflection, year):
+        self.language = language
```
Comment (BenChisnell) [inline prac_06/programming_language.py:14]: Missing docstring

File: prac_06/programming_language.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+        self.typing = typing
+        self.reflection = reflection
+        self.year = year
+
+    def is_dynamic(self):
+        if self.typing == "Dynamic":
```
Comment (BenChisnell) [inline prac_06/programming_language.py:21]: Missing docstring

File: prac_06/guitar.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+Estimated Time: 30 minutes
+Start Time: 7:24 AM
+End Time: 7:50 AM, 26 minutes
+
+This module contains the Guitar class to represent a guitar with its name,
+year of manufacture, and cost. It includes methods for getting the guitar's age
+and determining if it is vintage (50 years or older).
+"""
+class Guitar:
+    def __init__(self, name= "", year=0, cost=0):
```
Comment (BenChisnell) [inline prac_06/guitar.py:14]: Missing docstring

File: prac_06/guitar.py [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+        self.name = name
+        self.year = year
+        self.cost = cost
+
+    def get_age(self):
+        current_year = 2024
```
Comment (BenChisnell) [inline prac_06/guitar.py:21]: Missing docstring

File: prac_06/guitar.py (hunk 1 above)
Comment (BenChisnell) [inline prac_06/guitar.py]: Missing docstring

**Review by BenChisnell — COMMENTED** [review]
Just a few missing docstrings and a while loop that doesn't work, but other than that, good work!

File: prac_06/guitars.py (hunk 2 above)
Comment (bipinsubedi4) [reply prac_06/guitars.py]: Thanks for your feedback but it will work mate as it says to run the loop until it exists

**Comment by bipinsubedi4** [comment]
Thank you for the response. Everything rectified. Thanks :)

**Pull request merged**